pkg/versions/git.go

```diff
@@ -23,6 +25,11 @@ func Git(remoteURL string, auth transport.AuthMethod) ([]*semver.Version, error)
 	if err != nil {
 		return nil, fmt.Errorf("git remote: %w", err)
 	}
+	if githubToken := os.Getenv("GITHUB_TOKEN"); githubToken != "" {
```

Comment:
could you move this to the main, and there set it on the `updatesClient` struct? like this

```go
updatesClient.GitAuth = &http.BasicAuth{
    Username: githubToken,
}
```

otherwise we might run into surprises later on, if someone were to add more auth options (GITHUB_TOKEN would always override those if we set it here)